plugins/module_utils/job.py

```diff
@@ -76,9 +76,47 @@ def _get_job_output(job_id="*", owner="*", job_name="*", dd_name=""):
                 str(rc), str(err)
             )
         )
-    if not out:
-        raise RuntimeError("Failed to retrieve job output. No job output found.")
-    jobs = _parse_jobs(out)
+    jobs = []
+    if out:
+        jobs = _parse_jobs(out)
+    if not jobs:
+        jobs = _job_not_found(job_id, owner, job_name, dd_name)
+
+    return jobs
+
+
+def _job_not_found(job_id, owner, job_name, dd_name):
+    jobs = []
+
+    job = {}
+
+    job["job_id"] = job_id
+    job["job_name"] = job_name
+    job["subsystem"] = "NOTFOUND"
```

Comment:
When the job string is passed in for a job that does not exist , I wonder if any of these `NOTFOUND` could be populated. Its likely you won't have that info but I am unsure without trying it out. 
Ie subsystem & system

Reply:
This code is triggered when there is no return for the given JobID.